examples/pytorch/graphsaint/train_full.py

```diff
@@ -0,0 +1,168 @@
+    for epoch in range(args.n_epochs):
+        model.train()
+        # forward
+        pred = model(train_g)
+        train_labels = train_g.ndata['label']
+
+        if multitask:
+            loss = F.binary_cross_entropy_with_logits(pred, train_labels)
+        else:
+            loss = F.cross_entropy(pred, train_labels)
+
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+        print(f"epoch:{epoch}/{args.n_epochs}: training loss", loss.item())
+
+        # evaluate
+        if epoch % args.val_every == 0:
+            val_f1_mic, val_f1_mac = evaluate(
+                model, g, labels, val_mask, multitask)
+            print(
+                "Val F1-mic {:.4f}, Val F1-mac {:.4f}".format(val_f1_mic, val_f1_mac))
+            if val_f1_mic > best_f1:
+                best_f1 = val_f1_mic
+                print('new best val f1:', best_f1)
+                torch.save(model.state_dict(), os.path.join(
+                    log_dir, 'best_model.pkl'))
+
+    end_time = time.time()
+    print(f'training using time {end_time - start_time}')
+
+    # test
+    if True:
+        model.load_state_dict(torch.load(os.path.join(
+            log_dir, 'best_model.pkl')))
+    test_f1_mic, test_f1_mac = evaluate(
+        model, g, labels, test_mask, multitask)
+    print("Test F1-mic{:.4f}, Test F1-mac{:.4f}".format(test_f1_mic, test_f1_mac))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='GCN')
+    # data source params
+    parser.add_argument("--dataset", type=str, choices=['ppi', 'flickr', 'reddit', 'yelp', 'amazon'], default='ppi',
+                        help="Name of dataset.")
+    # cuda params
+    parser.add_argument("--gpu", type=int, default=-1,
+                        help="GPU index. Default: -1, using CPU.")
+    # model params
+    parser.add_argument("--n-hidden", type=int, default=512,
+                        help="Number of hidden gcn units")
+    parser.add_argument("--arch", type=str, default="1-0-1-0",
+                        help="Network architecture. 1 means an order 1 layer (self feature plus 1-hop neighbor "
+                             "feature), and 0 means an order 0 layer (self feature only)")
+    parser.add_argument("--dropout", type=float, default=0,
+                        help="Dropout rate")
+    parser.add_argument("--batch-norm", action='store_true',
+                        help="Whether to use batch norm")
+    # training params
+    parser.add_argument("--n-epochs", type=int, default=100,
+                        help="Number of training epochs")
+    parser.add_argument("--lr", type=float, default=0.01,
+                        help="Learning rate")
+    parser.add_argument("--weight-decay", type=float, default=0,
+                        help="Weight for L2 reg")
+    parser.add_argument("--val-every", type=int, default=1,
+                        help="Number of epoch of doing inference on validation")
```

Comment:
"Number of epoch of doing inference on validation" -> "Frequency of evaluation on the validation set in number of epochs"